src/main/java/com/redislabs/redisai/Model.java

```diff
@@ -0,0 +1,152 @@
+package com.redislabs.redisai;
+
+import com.redislabs.redisai.exceptions.JRedisAIRunTimeException;
+import redis.clients.jedis.util.SafeEncoder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Direct mapping to RedisAI Model
+ */
+public class Model {
+    private Backend backend;
+    private Device device;
+    private String[] inputs;
+    private String[] outputs;
+    private byte[] blob;
+    private String tag;
+
+    /**
+     * @param backend - the backend for the model. can be one of TF, TFLITE, TORCH or ONNX
+     * @param device  - the device that will execute the model. can be of CPU or GPU
+     * @param inputs  - one or more names of the model's input nodes (applicable only for TensorFlow models)
+     * @param outputs - one or more names of the model's output nodes (applicable only for TensorFlow models)
+     * @param blob    - the Protobuf-serialized model
+     */
+    public Model(Backend backend, Device device, String[] inputs, String[] outputs, byte[] blob) {
+        this.backend = backend;
+        this.device = device;
+        this.inputs = inputs;
+        this.outputs = outputs;
+        this.blob = blob;
+        this.tag = "";
+    }
+
+    public static Model createModelFromRespReply(List<?> reply) {
+        Model model = null;
+        Backend backend = null;
+        Device device = null;
+        String tag = null;
+        byte[] blob = null;
+        for (int i = 0; i < reply.size(); i += 2) {
+            String arrayKey = SafeEncoder.encode((byte[]) reply.get(i));
+            switch (arrayKey) {
+                case "backend":
+                    String backendString = SafeEncoder.encode((byte[]) reply.get(i + 1));
+                    backend = Backend.valueOf(backendString);
+                    if (backend == null) {
+                        throw new JRedisAIRunTimeException("Unrecognized backend: " + backendString);
+                    }
+                    break;
+                case "device":
+                    String deviceString = SafeEncoder.encode((byte[]) reply.get(i + 1));
+                    device = Device.valueOf(deviceString);
+                    if (device == null) {
+                        throw new JRedisAIRunTimeException("Unrecognized device: " + deviceString);
+                    }
+                    break;
+                case "tag":
+                    tag = SafeEncoder.encode((byte[]) reply.get(i + 1));
+                    break;
+                case "blob":
+                    blob = (byte[]) reply.get(i + 1);
+                    break;
+                default:
+                    break;
+            }
+        }
+        if (backend != null && device != null && blob != null) {
+            model = new Model(backend,
+                    device,
+                    new String[0], new String[0], blob);
```

Comment:
@filipecosta90 don't we have the inputs and outputs?

Reply:
that will be added in https://github.com/RedisAI/RedisAI/issues/378 ( doing it today )
I can already prepare the INPUTS,OUTPUTS, BATCHSIZE, and MINBATCHSIZE

Reply:
solved in the latest commit